### PR TITLE
adds a new top-level cmakelists + changes the install dir of systlogs…

### PR DESCRIPTION
**Pull request by JoelStienlet**
- creates a new cmakelists in the top-level directory
- modifies the build & test script to work with the new cmakelists, added a few usage comments
- modifies the location where the headers of syslog get installed. It was not consistent with logrotate. Now both install their header in include/g3sinks/

### CODE REVIEW

File: scripts/buildAndRunTests.sh [hunk 1]
```diff
@@ -1,33 +1,82 @@
...
-cd 3rdparty && unzip gtest-1.7.0.zip
-cd $pwd
+mkdir build 
 
-#logrotate
-echo "BUILD LOGROTATE SINK AND RUN TESTS"
-cd logrotate && mkdir build && cd build && cmake -DADD_LOGROTATE_UNIT_TEST=ON ..
-make -j && ./UnitTestRunner
-echo "FINISHED LOGROTATE BUILD AND TESTS"
+if [ $do_tests == true ]; then
```
Comment (KjellKod) [inline scripts/buildAndRunTests.sh]: Why is this needed?  If one runs the `buildAndrunTests.sh` scripts I think it's assumed that you will build the tests, right?

File: syslog/src/syslogsink.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #include "g3log/logmessage.hpp"
-#include "g3log/syslogsink.hpp"
+#include "g3sinks/syslogsink.hpp"
```
Comment (KjellKod) [inline syslog/src/syslogsink.cpp]: nice catch

File: scripts/buildAndRunTests.sh (hunk 1 above)
Comment (JoelStienlet) [reply scripts/buildAndRunTests.sh]: Actually I found the script convenient to debug the cmake files with different options (shared or static), it's also useful to check for syntax errors etc... So I found it convenient to have an option to disable the tests, and also disable installing g3log.
This is by no means a required change, only a small upgrade to the script that I found convenient, and I though it could come in handy to others too.

Comment (JoelStienlet) [reply scripts/buildAndRunTests.sh]: I've removed the option to skip the tests, I wait for this to be merged before I submit another PR for issue #78

Comment (JoelStienlet) [reply scripts/buildAndRunTests.sh]: (Finally I didn't wait for this to be merged to submit the other PR, but the two PRs are really independent, so currently I've two PRs open in g3sinks and one open in g3log )
